### PR TITLE
csharp: add test file in test/lsp-csharp-test.el

### PR DESCRIPTION
**Pull request by razzmatazz**
Adds a single test for csharp, so we can check if workspace initializes on a sample project.

### CODE REVIEW

File: test/lsp-csharp-test.el [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+(ert-deftest lsp-clangd-initialised-workspace ()
+  (skip-unless (memq system-type '(gnu/linux darwin windows-nt)))
+  (lsp-in-sample-csharp-project
+   (->
+    ;; now check that the workspace has started
+    (lsp-test-wait (eq 'initialized
+                       (lsp--workspace-status (cl-first (lsp-workspaces)))))
+   (deferred:sync!))
+   ))
```
Comment (ericdallo) [inline test/lsp-csharp-test.el:67]:
```suggestion
   (deferred:sync!))))
```

**Comment by razzmatazz** [comment]
@jcs090218 could you guide me a little bit into how I can make lsp-csharp-test.el to be run with ert?

Also, do I have to install the server myself somehow during the test setup process or will the automatic server installation work here? Also, do I need to add hooks or something so that `(find-file (f-join lsp-test-location "fixtures/SampleCSharpProject/Program.cs"))` will actually start lsp mode?

I copied most of the code from lsp-clangd-test.el, but I might not know the underlying context well here, on how integration tests work..

**Comment by jcs090218** [comment]
> @jcs090218 could you guide me a little bit into how I can make lsp-csharp-test.el to be run with ert? It seems like CI skips lsp-csharp-test.el and there is no registration list for tests so I figure those are autodetected somehow?

Sorry for the late reply! I can see the test on Windows but not on Unix system. I think the culprit is the line,

https://github.com/emacs-lsp/lsp-mode/blob/5ca68ed5b9295f298fa45e3cc0f4901a7c471c46/Makefile#L96

> Also, do I have to install the server myself somehow during the test setup process or will the automatic server installation work here? Also, do I need to add hooks or something so that (find-file (f-join lsp-test-location "fixtures/SampleCSharpProject/Program.cs")) will actually start lsp mode?

No, I think you would have to install the server in the test file. Is there a function so we can install the server? Maybe like `(lsp-install-server 'csharp)` so it would be a lot easier even with other test cases? You would also need to install `csharp-mode`.

**Comment by jcs090218** [comment]
I think the function `lsp-install-server` now support argument `server-id`. You can now install the server using this function! Thanks to @yyoncho in #2817.

**Pull request closed without merging**